### ISSUE TITLE
Questions about the exact hyperparameters for DIGER (SDUD + FrqUD) on Beauty/B-Shop

### ISSUE THREAD

**Issue by xiaoxiao735** [junchen-fu/DIGER#5]
Dear junchen-fu,

Thank you very much for releasing the code and checkpoints for this work.

I am currently reproducing the DIGER (SDUD + FrqUD) result on the Beauty/B-Shop dataset. My evaluation of the released `beauty_both_epoch91.pt` checkpoint closely matches the result .

Therefore, the dataset and evaluation pipeline appear to be correct.

However, when I train the SDUD+FrqUD configuration from scratch, my best result is lower:

| Setting | Recall@5 | Recall@10 | NDCG@5 | NDCG@10 |
| My reproduction | 0.041989 | 0.063274 | 0.027678 | 0.034514 |

My current training configuration is as follows.

### Hardware and batching

- GPU: 1 × NVIDIA RTX A6000 (48 GB)
- Per-device batch size: 32
- Gradient accumulation steps: 8
- Effective global batch size: 256
- Evaluation batch size: 32
- Number of beams: 20

### Model configuration

- Codebook size: 256
- Semantic ID length: 3, plus one conflict token
- Encoder layers: 6
- Decoder layers: 6
- Hidden size (`d_model`): 128
- Feed-forward dimension (`d_ff`): 512
- Attention heads: 4
- Dropout: 0.1
- Maximum history length: 50

### Optimization

- Optimizer: AdamW
- Recommender learning rate: 0.001
- ID tokenizer learning rate: 0.00001
- Weight decay: `0.05`
- Learning-rate scheduler: cosine
- Warm-up steps: 500
- Maximum epochs: 120
- Early-stopping patience: 15
- Validation metric: NDCG@10
- Random seed: 2020

### Gumbel and uncertainty-decay configuration

- Strategy: SDUD + FrqUD
- Gumbel temperature: `2.0`
- Noise type: Gumbel
- Initial noise standard deviation: `1.0`
- Learnable sigma: enabled
- Sigma learning rate: `1e-3`
- `sigma_lambda`: `1.8`
- Simple uncertainty loss corresponding to Eq. (9): enabled
- Adaptive frequency-based selection: enabled
- Hot-code threshold ratio `r`: `1.5`
- Gumbel hard-switch epoch: `0`
- Temperature annealing: disabled
- Dynamic sigma learning rate: enabled

For the FrqUD EMA coefficient, I followed the paper and used:

- Usage EMA coefficient `β` / `usage_momentum`: `0.25`

However, the current implementation appears to use `usage_momentum=0.99` in some configurations or as the default value. I am unsure whether the `β=0.25` stated in the paper corresponds exactly to the `usage_momentum` parameter used by the released implementation.

### Loss weights and freezing settings

- Generation/code loss weight: `1.0`
- Reconstruction loss weight: `1.0`
- VQ loss weight: `1.0`
- QS loss weight: `0`
- Balance loss weight: `0`
- Gate loss weight: `0`
- Semantic embedding: frozen
- ID encoder: frozen initially, with `freeze_id_epochs=0`
- ID decoder: frozen
- RQ module: not frozen

Could you please help clarify the following points?

1. What was the exact value of `usage_momentum` used for the SDUD+FrqUD result : `0.25` or `0.99`?
2. Does the `β=0.25` described in Eq. (10) correspond directly to the `usage_momentum` variable in the code?
3。Could you please confirm whether the released `beauty_both_epoch91.pt` checkpoint was trained with `sigma_lambda=1.8`, `initial_std=1.0`, and `hot_threshold_ratio=1.5`? Since the paper provides only the search ranges rather than the final Beauty-specific values, I would greatly appreciate clarification on the exact settings used for the SDUD+FrqUD result in the paper.
4. Was the uncertainty loss for this experiment implemented exactly as Eq. (9), or was another sigma regularization formulation used?
5. The currently released Beauty configuration appears to use a batch size of 256, one gradient accumulation step, 120 maximum epochs, and an early-stopping patience of 15. However, these settings are not stored in the released checkpoint. Could you please confirm whether these were the exact effective training settings used to produce `beauty_both_epoch91.pt`?

I would be very grateful for any clarification. Thank you again for releasing the code and checkpoints.

**Comment by junchen-fu** [comment]
Hi @xiaoxiao735 

Thank you for your detailed report. I checked the original training log for `beauty_both_epoch91.pt`. The settings were:

- **Usage EMA:** `usage_momentum=0.99`. It multiplies the previous EMA value, following the coefficient convention in Eq. (10), with updates during training forwards. The separate YAML parameter `beta=0.25` controls the VQ commitment loss.
- **Uncertainty loss:** `use_simple_uncertainty_loss=false`. This run used the `AutoSigmaGumbel` regularization branch with `sigma_reg_weight=2.0`, rather than the simple loss in Eq. (9). `sigma_lambda` does not apply to this branch.
- **Noise and threshold:** `initial_std=1.0` and `hot_threshold_ratio=1.5`.
- **Training:** Batch size **256**, gradient accumulation **1**, maximum epochs **120**, and early-stopping patience **15**, using validation **NDCG@10** for checkpoint selection.

With the unmodified released configuration, the corresponding command is:

```bash
bash scripts/run_experiment.sh beauty both
```

Please first align the EMA and uncertainty-loss settings above when reproducing this checkpoint.

**Comment by xiaoxiao735** [comment]
Hi @junchen-fu,  
Thank you very much for checking the original training log and providing these detailed settings. Your clarification resolves my confusion, especially regarding the distinction between usage_momentum=0.99 and beta=0.25, as well as the uncertainty-loss branch used for the released checkpoint.  
I will align my configuration with these settings and rerun the Beauty both experiment. I really appreciate your time and help!
